### DISCUSSION TITLE
We have similar projects, :)

### DISCUSSION THREAD

**Discussion by RonIovine** [daniele77/cli#85] [General]
Hi Daniele, I came across this project & was very interested in it.  I have a very similar project with similar goals/use cases.  I like your header only approach & the use or C++14 and you are truly cross platform (*Nix & Windows).  One of my early implementations was C++, but I moved to a more strait C based API & implementation to maximize the possible uptake (it can safely be linked with C++ applications).  I also added multi-language support but my cross platform is for various flavors of *Nix.

Anyway, if you get time, checkout my repo & see what you think.  If you had any suggestions and/or comments I'd love to hear them.  If you think this post is inappropriate for this forum, feel free to delete/remove it.

https://github.com/RonIovine/pshell

Cheers,
Ron

**Comment by daniele77** [comment]
Hi @RonIovine 
I see the two libraries provide similar features, however, they have different targets: `cli` is more suitable for "modern C++" projects, while `pshell` is more oriented towards C projects and pre C++14 code.
Of course one can use a C library in a C++ codebase as well, but in my experience, C++ developers do prefer "modern C++ libraries" (when *good* C++ libraries do exist!) and on the other side, C developers don't really want to link a C++ library :-)
